CIP-_/README.md

```diff
@@ -0,0 +1,210 @@
+
+## Motivation: why is this CIP necessary?
+
+<b>Cardano needs to scale.</b> As the ecosystem grows and new users come in, blocks are regular at 100% load with 88KB per block allocation. Recently, events such as the Lil Sappys NFT mint have pushed the Cardano blockchain load to over 100% for several hours, preventing anyone from using the chain and providing an awful user experience to new users who complain on social media and give the ecosystem a bad reputation.
+
+<b>This can't continue if Cardano is to succeed.</b>
+
+Currently the existing <a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a> for coin selection is not sufficient as it was developed in a time before native assets and smart contracts were live on the Cardano Blockchain. Many existing dapps are using extremely poor coin selection resulting in transactions that are 2,3, sometimes 5 times larger then neccesary while also creating outputs that are difficult to handle in future coin selections.
+
+This CIP aims to provide at least 1 algorithm and open the conversation for multiple algorithms to exponentially improve coin selection which will decrease transaction sizes accross the board and increase Cardano's throughput.
+
+Optimizations are priorized in this order for this CIP:
+
+-   Transaction size reduction.
+-   Change Output size reduction for future transactions.
+-   Dust manage to prevent large small dust Utxos from piling up.
+
+## Specification
+
+### Scope
+
+<a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a> explains the basics of coin selection, so I won't repeat those here. Instead I will simply present an algorithm for optimizing coin selection. This algorithm will include optimizing the standard coin selection with change selection while also containing suggestions for optimizing collateral selection and optimizing smart contract transactions.
+
+This algorithm will be presented with the following composable structure, allowing for coin and change selection strategies to be composed together:
+
+-   Aggregator Function
+-   Input Selection Function
+-   Change Selection Function
+
+### Algorithms
+
+Random improve was an original algorithm presented in <a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a>, this algorithm had a series of good ideas that can be expanded upon here. The first proposed algorithm will simply be an Optimized Random Improve implementation.
+
+#### Optimized Random Improve
+
+Unlike random improve which operates on a per output basis. This optimized random improve algorithm will operate on a PER ASSET basis. Meaning the algorithm will be designed to operate on and select inputs that satisfy values for the various native assets in this transaction.
+
+-   <b>Step 1: Aggregator Function</b>
```

Comment:
```suggestion
##### Step 1: Aggregator Function
```

Reply:
Use of headings rather than HTML "bold" tags for at least the top-level steps will allow better and easier permalinking to specific sections of the final document